src/FSharpVSPowerTools.Logic/SyntaxConstructClassifier.fs

```diff
@@ -175,29 +175,53 @@ type SyntaxConstructClassifier
         else Some()
 
     let mergeSpans (oldSpans: CategorizedColumnSpan<_>[]) (newSpans: CategorizedColumnSpan<_>[]) =
+        let getTypeCheckerSpans spans =
+            spans
+            |> Array.filter (fun x -> 
+                match x.Category with
+                | Category.Escaped 
+                | Category.Operator
+                | Category.Other
+                | Category.Printf
+                | Category.Quotation 
+                | Category.Unused -> false
+                | _ -> true) 
+
         let getLineRange spans =
-            let lines = 
-                spans 
-                |> Array.filter (fun x -> 
-                    match x.Category with
-                    | Category.Escaped 
-                    | Category.Operator
-                    | Category.Other
-                    | Category.Printf
-                    | Category.Quotation 
-                    | Category.Unused -> false
-                    | _ -> true) 
-                |> Array.map (fun x -> x.WordSpan.Line)
-            Array.min lines, Array.max lines
+            let typeCheckerSpans = getTypeCheckerSpans spans
+            typeCheckerSpans,
+            typeCheckerSpans
+            |> Array.map (fun x -> x.WordSpan.Line)
+            |> function [||] -> -1, -1 | lines -> Array.min lines, Array.max lines
```

Comment:
This `-1` thing is convenient but dangerous. There is no tracking from the type system so it's easy to misuse it.